### DISCUSSION TITLE
Could I build a nested `Pick` from dot path notation with Hotscript?

### DISCUSSION THREAD

**Discussion by nandorojo** [gvergnaud/HOTScript#73] [Q&A] [answered]
Hey there, super excited about Hotscript. Thanks so much for sharing it @gvergnaud.

I have an object that looks like this:

```ts
type Filter = {
  booking: {
    startTime: { id: string }
  },
  // other types here...
}
```

On another type, I have the nested dot path:

```ts
const startTimePath = 'booking.startTime' satisfies DotPath<Filter>
```

I now want to "unfurl" these back an object, where I give a dot path and a value, and it constructs that object for me:

```ts
type Sort =  'desc' | 'asc'

const startTimePath = 'booking.startTime' satisfies DotPath<Filter>

// something like DotPathKey is what I'm looking for
const order: DotPathKey<Filter, typeof startTimePath, Sort> = {
  booking: {
    startTime: 'asc'
  },
}
```

If I weren't using dot paths, this would be easy, since I could just use `keyof`:

```ts
const order: { [key in keyof Filter]: Value } = {
  booking: { sort: 'desc' }
}
```

**Comment by gvergnaud** [comment, marked as answer]
Hey! I think you are looking for the `Object.Get` or the `Objects.Update` function: [Playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wBJwGZQiOAcgAsIZUBjKYMGQgKEZgE8wBTOAMWABsY2ocALxwA3nABGECAGtgAOwDmALjFxUMZLAAqoNqvHAAJqo3UlcAL5WrTVhyhtUARmGYAdAFEAbsh4AeegB6ILgwuAA9AH56MIx3AHkJACs2CnJ3AFUwI2R+fwAiKVkFRXcNLRhdEDYCgBo4ArQKArgAH0ajJxaG7j4BAD56AYBuO3Y4R1QAJjd4nz9AkPDImLjElLSMgHE2GELiuSVyzR09eq5efigh0aA)

```ts
import * as H from 'hotscript'


type Filter = { booking: { startTime: { id: string } } }

type res1 = H.Eval<
//    ^? { booking: { startTime: "asc" | "desc" } }
  H.Objects.Update<"booking.startTime", "asc" | "desc", Filter>
>;

type res2 = H.Eval<
//    ^? { id: string }
  H.Objects.Get<"booking.startTime", Filter>
>;
```
